src/System.Management.Automation/engine/ExperimentalFeature/EnableDisableExperimentalFeatureCommand.cs

```diff
@@ -112,26 +113,28 @@ public class ExperimentalFeatureNameCompleter : IArgumentCompleter
         /// <param name="commandAst">The command AST.</param>
         /// <param name="fakeBoundParameters">The fake bound parameters.</param>
         /// <returns>List of Completion Results.</returns>
-        public IEnumerable<CompletionResult> CompleteArgument(string commandName, string parameterName, string wordToComplete, CommandAst commandAst, IDictionary fakeBoundParameters)
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
         {
-            if (fakeBoundParameters == null)
-            {
-                throw PSTraceSource.NewArgumentNullException(nameof(fakeBoundParameters));
-            }
-
-            var commandInfo = new CmdletInfo("Get-ExperimentalFeature", typeof(GetExperimentalFeatureCommand));
-            var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace)
-                .AddCommand(commandInfo)
-                .AddParameter("Name", wordToComplete + "*");
+            SortedSet<string> expirmentalFeatures = new(StringComparer.OrdinalIgnoreCase);
 
-            HashSet<string> names = new HashSet<string>();
-            var results = ps.Invoke<ExperimentalFeature>();
-            foreach (var result in results)
+            foreach (ExperimentalFeature feature in GetExperimentalFeatures())
             {
-                names.Add(result.Name);
+                expirmentalFeatures.Add(feature.Name);
             }
 
-            return names.Order().Select(static name => new CompletionResult(name, name, CompletionResultType.Text, name));
+            return CompletionCompleters.GetMatchingResults(wordToComplete, expirmentalFeatures);
```

Comment:
@iSazonov Do you think `GetMatchingResults` should eventually be in a base class? Seems like not ideal design to keep calling static method in `CompletionCompleters` for matching results, seems much better placed in a base class since this now is used in a lot of argument completers in this repo.

I think CompletionCompleters is getting too big and probably need to start abstracting common code for argument completers. 

Let me know your thoughts. 🙂.

Reply:
I know this code is big and confusing, but I've never analyzed it as a whole. Considering that this is all about tree analysis, perhaps we could think about more unified code, code generation, and automated test generation. Although maybe it's not worth it. I don't know. In any case, there's probably something to put in better order.

Reply:
I think I already mentioned this, I would like to understand if this code is related to the internal API. If not, why is it in the engine - perhaps it should be in the host, but at the same time be present in the SDK.

Reply:
I agree. I think I will see if base class can be moved to SDK where this code can be exposed easier. Maybe after this PR I will try this and see how it goes 🙂.

In its current state I don't think it belongs in engine.